File: 45.jump-game-ii/solution.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        maxJmp = 0
        nextMaxJmp = 0

        N = len(nums)
        if (N == 1):
            return 0
        cnt = 0

        for i in range(N):

            if (nextMaxJmp < i + nums[i]):
                nextMaxJmp = i + nums[i]

            if (i == maxJmp):
                maxJmp = nextMaxJmp
                cnt += 1
                if (maxJmp >= N - 1):
                    return cnt

File: 45.jump-game-ii/solution.kt
class Solution {
    fun jump(nums: IntArray): Int {
        if (nums.size == 1) {
            return 0
        }

        var cnt = 0
        var nextMaxJmp = 0
        var maxJmp = 0
        for (i in 0 until nums.size) {
            if (nextMaxJmp < i + nums[i]) {
                nextMaxJmp = i + nums[i]
            }
            if (maxJmp == i) {
                maxJmp = nextMaxJmp
                cnt += 1
                if (maxJmp >= nums.size - 1) {
                    return cnt
                }
            }
        }
        return cnt
    }
}
